From 3ddd9504cf8e37d73a7f927f1e557650d63e0aee Mon Sep 17 00:00:00 2001
From: Nathaniel Russell <46272571+nater1983@users.noreply.github.com>
Date: Thu, 25 Jan 2024 19:54:47 -0600
Subject: [PATCH] Update meson.build

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index c46f11b..7b0b2a9 100644
--- a/meson.build
+++ b/meson.build
@@ -1,4 +1,4 @@
-project('webp-pixbuf-loader', 'c', meson_version: '>=0.54')
+project('webp-pixbuf-loader', 'c', version: '0.2.6', meson_version: '>=0.54')
 cc = meson.get_compiler('c')
 gio = dependency('gio-2.0', method: 'pkg-config')
 gdkpb = dependency('gdk-pixbuf-2.0', version: '>2.22.0', method: 'pkg-config')
